### DISCUSSION TITLE
Custom Alarm Panel Card

### DISCUSSION THREAD

**Discussion by crumbliness** [Dilbert66/esphome-vistaECP#34] [Q&A] [answered]
I have everything set up and running on a Vista 15P with ESPHome except the new alarm panel. Thanks for a great project!

Home Assistant is showing error messages when a new custom card is created. I've copied lines 1 thru 76 of the lovelace.yaml file to the HA card configuration.

The first 2 errors are 'You need to define a kpd_line1' and 'You need to define a kpd_line2'. Lines 4 and 5 of lovelace.yaml are 'disp_line1' and 'disp_line2'. Should these be changed to 'disp_line1' and 'disp_line2'?

Next error is 'You need to define a keypad service'. Isn't the service defined in line 8? What am I missing here?

![image](https://user-images.githubusercontent.com/57737088/120087688-13955000-c0b8-11eb-98f4-01d2bb90e2d3.png)

**Comment by Dilbert66** [comment]
Looks like you have a cached older version of the card js file. I updated the card and changed some variables.  Reload the js file.  Sometimes with HA it won't update the code unless you force it.  I do it by adding a url variable to the js path and change it everytime.
![image](https://user-images.githubusercontent.com/7193213/120088027-d1214280-c0ba-11eb-85de-ff700090d2de.png)

**Comment by crumbliness** [comment]
Alain - Thanks for the prompt reply. I downloaded the files again from
github even though it doesn't look like they have been updated - maybe it's
something that I don't understand about github.

I reloaded the alarm-keypad-card.js (tried both copies) and saw no change.
The alarm-keypad-card.js files downloaded today are Version 0.0.4a - same
version as those I downloaded on 28May21.

Do I need to wait for github to refresh?

**Comment by crumbliness** [comment]
Alain - Just re-did everything from the start and the alarm keypad card
appears - Thanks - The alarm stay is not working for me at the moment but I
need to dig in a little deeper to see what's up.

**Reply by appleguru** [reply]
I assume you put your own arm code in correctly in the card config?

**Comment by Dilbert66** [comment]
I'm assuming here that you are using mqtt and not esphome.  Without knowing how you configured the homeassistant services, it's hard to know what the issue is.

**Comment by crumbliness** [comment]
I have been using ESPHome and the HA  Alarm Control Card since Jan21. The
HA Alarm Control Card works for 'Arm Home', 'Arm Away' and 'Disarm'
commands although there are error messages in the log files showing that
card doesn't like 'Arm Home' and would prefer 'Arm Stay' instead. I've been
ignoring these since it works.

Earlier this week, I updated my interface board replacing R1 and uploading
the current VistaAlarm.yaml - all of that went well,I think. The new Alarm
Panel Card is where I stumbled.

My arm code is correct in the ESPHome secrets.yaml file. And that works
with the existing HA  Alarm Control Card for 'Arm Home', 'Arm Away' and
'Disarm' commands. If there is a different card config for the Custom Alarm
Panel Card, I haven't found it yet. I am still using the Sample Home
Assistant Template Alarm Control Panel configuration verbatim from the
Readme.md file in my HA configuration.yaml.

When using the Custom Alarm Panel Card, armcode + AWAY only produces a
single beep on my 6160RF panel. Same with armcode + Stay. However,
armcode + DISARM does disarm an armed system.

**Comment by Dilbert66** [comment]
That's the issue then.  The cmds for a,b,c,d need to be customized to your own code. For the lines below, replace 1234 with your own arm/disarm code.  I should clarify that in the docs.

```
For the MQTT example:

cmd_A:  topic: vista/Set/Cmd  payload: '!12343'
cmd_B:  topic: vista/Set/Cmd  payload: '!12342'
cmd_C:  topic: vista/Set/Cmd  payload: '!12341'
cmd_D:  topic: vista/Set/Cmd  payload: '!12346#'


```
```
For the ESPHOME example:

cmd_A:  keys: '12343'
cmd_B:  keys: '12342'
cmd_C:  keys: '12341'
cmd_D:  keys: '12346#'
```

**Comment by Dilbert66** [comment]
One clarification.  The alarm code setup in the esphome yaml only gets used if you send a command without an access code.  With the commands setup in the alarm panel code, I used the raw commands (as if you were entering from the keypad)  using a sample disarm/arm code of 1234 as an example.    You can instead program the key to send an 'A" or a 'S' instead of the raw command '12343' or '12342'.   When the esp gets an 'A' or 'S' , it will then use the default code programmed in the yaml in response to the access code request if needed.   I'm used to sending direct commands instead of the shortcuts.

**Comment by crumbliness** [comment]
Well, That was easy! Everything is working now. Thanks!

Would deleting the alarm-keypad-card,.js file in the root directory be a
good idea? The updated version is in the ha_cards sub directory.

**Reply by crumbliness** [reply]
I'd marked this one as answered - if I only knew how ...

**Reply by Dilbert66** [reply]
As long as you have the js file in the www subdirectory of your home assistant config

**Comment by Dilbert66** [comment, marked as answer]
Ok, i see the issue. That was my fault. I had an old copy in the root of the repository. Sorry about that. I will remove and push an update after I fix another isssue with RF
